https.c

```diff
@@ -349,7 +358,7 @@ https_init(const char *useragent, const char *cafile, const char *proxy)
                         return (HTTPS_ERR_LIB);
                 }
         }
-        if ((ctx->ssl_ctx = SSL_CTX_new(TLSv1_client_method())) == NULL) {
+        if ((ctx->ssl_ctx = SSL_CTX_new(SSLv23_client_method())) == NULL) {
```

Comment:
Per the discussion in the PR, I think this is supposed to be `TLS_client_method`. Also, you mentioned that `TLSv1_client_method` is deprecated, but looking at the [man page](https://www.openssl.org/docs/man1.1.0/man3/TLSv1_client_method.html) I don't see that. Could you point me in the right direction?

Ultimately I think `TLS_client_method` is the best, most general solution. If OpenSSL is compiled with `OPENSSL_NO_TLS1_METHOD` we won't have `TLSv1_client_method` available.

Reply:
Hi. Compiling with `TLSv1_client_method` on Ubuntu 18.04 (OpenSSL 1.1.0g) results in this:

```
https.c: In function ‘https_init’:
https.c:361:9: warning: ‘TLSv1_client_method’ is deprecated [-Wdeprecated-declarations]
         if ((ctx->ssl_ctx = SSL_CTX_new(TLSv1_client_method())) == NULL) {
         ^~
In file included from /usr/include/openssl/ct.h:13:0,
                 from /usr/include/openssl/ssl.h:61,
                 from https.c:26:
/usr/include/openssl/ssl.h:1626:1: note: declared here
 DEPRECATEDIN_1_1_0(__owur const SSL_METHOD *TLSv1_client_method(void)) /* TLSv1.0 */

```
I would agree that `TLS_client_method` seems to be the way to go. I don't see any mention of deprecation for `TLSv1_client_method` so that's odd. The page *does* mention deprecation of `SSLv23_client_method()` so maybe that should be reconsidered:

>... and SSLv23_client_method() were deprecated and the preferred TLS_method(), TLS_server_method() and TLS_client_method() functions were introduced in OpenSSL 1.1.0.

There is an older apache bug report [here](https://bz.apache.org/bugzilla/show_bug.cgi?id=60111) noting the deprecation as well a PHP (2019-01-15) bug report [here](https://bugs.php.net/bug.php?id=77463).

Reply:
Since `sslv23_client_method` is still workable on the newer versions of OpenSSL I'm good to stick with this for now. It's what we use in Duo Unix currently.
Eventually using `TLS_client_method` is probably the way to go

Reply:
We do use `SSLv23_client_method` in Duo Unix, but we also blacklist SSLv2 and SSLv3: https://github.com/duosecurity/duo_unix/blob/master/lib/https.c#L406-L412

Let's do the same thing here :+1: